Userland/Libraries/LibCore/AnonymousBuffer.cpp

```diff
@@ -6,44 +6,16 @@
 
 #include <AK/Try.h>
 #include <LibCore/AnonymousBuffer.h>
+#include <LibCore/System.h>
 #include <LibIPC/File.h>
 #include <fcntl.h>
 #include <sys/mman.h>
 
-#if defined(__serenity__)
-#    include <serenity.h>
-#endif
-
-#if defined(__linux__) && !defined(MFD_CLOEXEC)
-#    include <linux/memfd.h>
-#    include <sys/syscall.h>
-
-static int memfd_create(const char* name, unsigned int flags)
-{
-    return syscall(SYS_memfd_create, name, flags);
-}
-#endif
-
 namespace Core {
 
 ErrorOr<AnonymousBuffer> AnonymousBuffer::create_with_size(size_t size)
 {
-    int fd = -1;
-#if defined(__serenity__)
-    fd = anon_create(round_up_to_power_of_two(size, PAGE_SIZE), O_CLOEXEC);
```

Comment:
You lost `round_up_to_power_of_two(size, PAGE_SIZE)`. Are you sure you don't want to round the size up to a multiple of the page size anymore?

Reply:
This is the reason the PR is on draft. Have patience.